dvc/cache/base.py

```diff
@@ -101,40 +101,52 @@ def changed(self, path_info, hash_info):
         Args:
             path_info: dict with path information.
             hash: expected hash value for this data.
+            filter_info: an optional argument to target a specific path.
 
         Returns:
             bool: True if data has changed, False otherwise.
         """
 
+        hash_origin = path_info
+        if filter_info:
+            path = filter_info
+            dir_info = self.get_dir_cache(hash_info)
+            hash_key = path.relative_to(path_info).parts
+            if hash_key in dir_info.trie:
+                hash_info = dir_info.trie[hash_key]
```

Comment:
Are you sure this works correctly for a subdirectory?

Reply:
Looks like this will be skipped for a subdir, right? But you can actually compute the hash for the subdir if you have hashes for all the files inside of it. Take a look at how we compute hash_info for the directory.

Reply:
I tried to address this problem in the 1ced36c, would you mind checking the latest revision out?